test/e2e/clusterctl_upgrade_test.go

```diff
@@ -74,7 +74,7 @@ var _ = Describe("When testing clusterctl upgrades using ClusterClass (CAPV 1.10
 	}, WithIP("WORKLOAD_CONTROL_PLANE_ENDPOINT_IP"))
 })
 
-var _ = Describe("When testing clusterctl upgrades using ClusterClass (CAPV 1.9=>current, CAPI 1.6=>1.7) [ClusterClass]", func() {
+var _ = Describe("When testing clusterctl upgrades using ClusterClass (CAPV 1.9=>current, CAPI 1.6=>1.7) [supervisor] [ClusterClass]", func() {
 	const specName = "clusterctl-upgrade-1.9-current" // prefix (clusterctl-upgrade) copied from CAPI
 	Setup(specName, func(testSpecificSettingsGetter func() testSettings) {
 		capi_e2e.ClusterctlUpgradeSpec(ctx, func() capi_e2e.ClusterctlUpgradeSpecInput {
```

Comment:
Depending on if this runs a supervisor test, we have to pass `vm-operator` and `net-operator` at `InitWithRuntimeExtensionProviders.`.

To do that we should extend the `testSettings` struct ([here](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/82ec23c17f3ec94f1729fa5cb73daf3e7f8e0fbe/test/e2e/e2e_setup_test.go)) by string array which get's set to the correct value:

* in Govmomi mode: `[]string{}`
* in supervisor mode: `[]string{"vm-operator", "net-operator"}`

So we can use it and set it similar to the `testSpecificSettingsGetter().ClusterctlConfigPath`, as e.g.`testSpecificSettingsGetter().InitWithRuntimeExtensionProviders`

Reply:
@chrischdi Thanks for the hint! Just back to work on this, will take a look.